packages/localize-tools/src/formatters/xliff.ts

```diff
@@ -192,12 +192,20 @@ export class XliffFormatter implements Formatter {
     targetDocument?: XMLDocument
   ): string {
     const translationsByName = makeMessageIdMap(targetMessages);
-    const doc = targetDocument || this.initNewTargetDocument(targetLocale);
+    const doc = targetDocument || this.initNewTargetDocument();
     const indent = this.createIndentFunction(doc);
     const existingTranslationsByName = this.makeExistingTranslationsMap(doc);
-    const body = getOneElementByTagNameOrThrow(doc, 'body');
     const names = new Set();
-    for (const {name, contents: sourceContents, desc} of sourceMessages) {
+    for (const {name, contents: sourceContents, desc, file} of sourceMessages) {
+      const body = this.getOrCreateFileBody(
+        doc,
+        targetLocale,
+        this.config.sourceLocale,
+        pathLib.join(
```

Comment:
there has to be a better way to get a nice path, but couldn't think of any relative path stuff. Maybe loop through all sourceMessages and see what the common root is.